### ISSUE TITLE
Transfer environment fails with Resource temporarily unavailable

### ISSUE THREAD

**Issue by aronpammer** [icecc/icecream#456]
I installed icecc on a macos and on a linux machine. I am using the 1.1 version of icecc, although 1.2 behaves the same way.

The scheduler runs on the macos machine. I started a daemon on the linux machine and the macos agent, both of them could successfully connect.

This is how I'm trying to build on the macos machine:
```
export ICECC_DEBUG=debug
export ICECC_TEST_REMOTEBUILD=1
export ICECC_CLANG_REMOTE_CPP=1
export ICECC_VERSION=x86_64:/path/to/tar.gz,Darwin17_x86_64:/path/to/tar.gz
/usr/local/bin/clang -target x86_64-apple-darwin16.0.0 -arch x86_64 <rest of the command>
```
(/usr/local/bin/clang is a symlink to icecc)

For some reason whenever I start this command I get the following:
On the macos:
```
ICECC[1771] 15:06:37: connected to /var/run/icecc/iceccd.socket
ICECC[1771] 15:06:37: env: x86_64 '/path/to/tar.gz'
ICECC[1771] 15:06:37: env: Darwin17_x86_64 '/path/to/tar.gz'
ICECC[1771] 15:06:37: preparing <.m file> to be compiled for Darwin17_x86_64
ICECC[1771] 15:06:37: Have to use host <ip>:10245 - Job ID: 13 - env: Darwin17_x86_64 - has env: false - match j: 0
ICECC[1771] 15:06:37: connected to <ip>
ICECC[1771] 15:06:38: <Transfer Environment>
ICECC[1771] 15:06:58: flush_writebuf() failed(Error: Resource temporarily unavailable)
ICECC[1771] 15:07:00: !wait_for_msg()
ICECC[1771] 15:07:00: write of source chunk to host <linux ip>
ICECC[1771] 15:07:00: failed (Error: Resource temporarily unavailable)
ICECC[1771] 15:07:00: </Transfer Environment: 22018ms>
ICECC[1771] 15:07:00: timeout <= 0
ICECC[1771] 15:07:00: !wait_for_msg()
ICECC[1771] 15:07:00: got exception Error 15 - write to host failed (<linux ip>)
```

On the linux node:
```
[14116] 15:16:04: handle_transfer_env
[14116] 15:16:34: handle_transfer_env_done
[14116] 15:16:34: exit code: 129
[14116] 15:16:34: installed_size: 0
[14116] 15:16:34: reannounce_environments
```

I tried the same thing but instead of a linux node I used another macos node, the same happened.

All of the ports are allowed, both machines are within the same network.

What am I missing here?

**Comment by llunak** [comment]
Should be hopefully fixed with the above commit.

**Comment by llunak** [comment]
Your network is too slow to send the compiler tarball within 20 seconds, or the tarball is too big. How big are those files you use for ICECC_VERSION?

**Comment by snake-scaly** [comment]
I'm having the same issue. The environment file in /var/cache/icecc/native/ is 22 megabytes. The broken pipe issue isn't consistent. That is it fails all the time currently but may work fine tomorrow. So slow network connection (VPN in my case) can be at fault.

Is this timeout enforced on the sending or on the receiving side? Is it possible to tweak it to allow for more time to transfer?

I tried 1.1, 1.2 and 1.3 with identical results. However I can only change versions on my machine. The scheduler and the worker nodes are out of my control.

**Comment by llunak** [comment]
As far as I can tell, this issue has been fixed for 1.3, and the problems were mostly on the receiving side. So you should preferably get the other nodes and the scheduler updated to 1.3. If you really can't do that, then set ICECC_SLOW_NETWORK=1 and that's probably all you can do locally.

**Comment by llunak** [comment]
Fixed in 1.3, nothing more to do here.
